### PR TITLE
Update glyph grid UI design

### PR DESCRIPTION
**Pull request by eliheuer**
Adds a new glyph box style and off-white background color
Removes the full-width baseline graphic
Spacing adjustments

See the macOS 11 screenshot below:
update-glyph-grid-ui-design

### CODE REVIEW

**Comment by alerque** [comment]
Is the size of the grid configurable?

**Comment by eliheuer** [comment]
Yes, by changing this line near the top of `src/widgets/grid.rs`:
```rust
const GLYPH_SIZE: f64 = 128.;
```

It would be nice if there was a window where people who don't want to rebuild a Rust program from source could make customizations like that though.

**Comment by alerque** [comment]
Roger that! The grid looks good by the way.

The need for runtime config has come up before. Sounds like a job for a different PR though.

Another eventual feature for this would be a fuzzy quick search to type in bits of the glyph id and jump to selecting it in the matrix. Also a job for another day.

**Review by alerque — APPROVED** [review]

**Comment by cmyr** [comment]
I'd like to do runtime theme reloading shortly, it shouldn't be very complicated and will make this kind of thing a lot easier to play around with.

In any case will give this a look through shortly, very happy to see the PR though this view has needed some love for a while.

File: src/widgets/editor.rs
Comment (cmyr) [inline src/widgets/editor.rs:237]: let's do `env.get(theme::GLYPH_LIST_BACKGROUND)`? Assuming these colors are intended to match..

File: src/theme.rs
Comment (cmyr) [inline src/theme.rs:35]: something to keep in mind is that this is also the color for selected text; if we want those to be different we could add a new constant, and use it *just* for the selected glyph background color.

**Review by cmyr — COMMENTED** [review]
This looks good, I'm happy to defer to your design decisions.

I'm happy to merge this as-is, but one thing I'll mention is that when we implement hot reloading for themes, we're going to want to use fewer 'magic numbers' for things like stroke width and colors, and move all those in to be named constants in the theme.

**Review by cmyr — APPROVED** [review]

**Comment by cmyr** [comment]
@eliheuer would you like me to fix this up and merge it?

**Comment by eliheuer** [comment]
@cmyr Thanks Colin, let me try to force push an updated making the requested changes, but after that or if I can't make it work, please fix it up and merge it. Thanks again!

File: src/widgets/editor.rs
Comment (eliheuer) [reply src/widgets/editor.rs]: I force pushed a new update (4d781ac) that fixes this. Thanks! Please feel free to make any needed changes before merging.

File: src/theme.rs
Comment (eliheuer) [reply src/theme.rs]: I force pushed a new update (4d781ac) that fixes this. I just used some temporary `Color::rgb8` colors for now. I didn't want to add any new constants since that will hopefully get worked out in a separate PR improving the theming system at some point. Please feel free to make any needed changes before merging.

**Comment by eliheuer** [comment]
I force pushed a new update (4d781ac) that fixes the issues above. Please let me know if there is anything I can do to help improve the theming system, I have free time this quarter and would be happy to work on moving that forward.

**Review by cmyr — APPROVED** [review]
Thanks!

**Pull request merged**
